lms/djangoapps/instructor_task/tasks_helper.py

```diff
@@ -1341,11 +1341,19 @@ def upload_proctored_exam_results_report(_xmodule_instance_args, _entry_id, cour
 def generate_students_certificates(
         _xmodule_instance_args, _entry_id, course_id, task_input, action_name):  # pylint: disable=unused-argument
     """
-    For a given `course_id`, generate certificates for all students
-    that are enrolled.
+    For a given `course_id`, generate certificates for only students present in 'students' key in task_input
+    json column, otherwise generate certificates for all enrolled students.
     """
     start_time = time()
     enrolled_students = CourseEnrollment.objects.users_enrolled_in(course_id)
+
+    instructor_task = InstructorTask.objects.get(pk=_entry_id)
+    task_input = json.loads(instructor_task.task_input or '{}')
```

Comment:
What is the reason for get `task_input` by querying `InstructorTask` instead of using `task_input` passed in arguments?